### PR TITLE
Implement google social connect, fixes #66

### LINKED ISSUE #66: Gmail social connect
I think I looked at that some time ago, but I abandoned. Reading the code again now, I can't remember why I did that... I think @GabrielAnca did a good job.

### PR DESCRIPTION
**Pull request by vendethiel**
Note to reviewer: Only the first commit is to review. The second one only has google's library in it.
I split the commits up so that you could get a readable diff.

Based on @GabrielAnca 's cool work :-).

I'm not happy with some of the workarounds I've had to put there, to be honest -- not at all.
The only other way I see would be a "google_register.php" file, that'd do what `shim_register_request` does, because Google doesn't let us forward GET query parameters.

Btw, I didn't change the migration file. Tell me if you want me to do it.

### CODE REVIEW

**Comment by GabrielAnca** [comment]
Hi!

Sorry I didn't have a lot of time to read the whole code, I hope to be able to do it soon :) It seems a really good job.

Anyway, I see you are using a vendor and that you included al lot of code from them. All those files are exposed to the users right now (I think), so I would better not include unnecessary files such as the examples folder. Other option is to set a deny from all htaccess in the includes folder (not sure if it would break anything, some testing would be needed).

I will have a look soon :) Regards!

**Comment by vendethiel** [comment]
Maybe even switching to another library would make sense... Not sure.

You're right though, of course. Thanks for your feedback.

**Comment by MightyGorgon** [comment]
@GabrielAnca, let me know when you have time to have a quick test/review! :-)

Thanks @vendethiel

**Comment by vendethiel** [comment]
no problem!

**Comment by vendethiel** [comment]
@GabrielAnca do you think you'll have time to review this, or are you too busy by now?

**Comment by vendethiel** [comment]
@MightyGorgon It's been more than a year, please consider merging

**Comment by MightyGorgon** [comment]
I was waiting for @GabrielAnca to review it... but I think he has no time.

I will merge the request next time I'll code something.

**Comment by vendethiel** [comment]
Hi MG, did you change the google image? IIRC mine is a bit too big. I can take a look later.

**Comment by MightyGorgon** [comment]
I sent you images a while ago... anyway, I'll take care of that.

File: includes/usercp_register.php [hunk 1]
```diff
@@ -537,7 +537,7 @@ function show_coppa()
 				}
 			}
 
-			$template->assign_vars(array(
+			$template->assign_vars($a=array(
```
Comment (MightyGorgon) [inline includes/usercp_register.php]: Whay "$a=" is needed for?

**Pull request merged**